crates/ibc/src/mock/context.rs

```diff
@@ -217,6 +219,131 @@ pub struct MockContext {
     pub logs: Vec<String>,
 }
 
+#[derive(Debug, TypedBuilder)]
```

Comment:
As the entire MockContext is subject to change soon, I didn't delve deeply into abstraction using `TypedBuilder` in this PR. Seeing as it works for our tests, I think it's fine. Though a question I got: couldn’t we directly derive `TypedBuilder` on the own `MockContext`?

Reply:
On board with this. Should preferably use existing solutions.
Let's discuss this in more detail later on to see how it would be best to derive Builders on different IBC types to keep things consistent and only for testing.